build-tools/build.sh

```diff
@@ -0,0 +1,223 @@
+    echo "\`\`\`" >>$GITHUB_STEP_SUMMARY
+    echo "$norm_output" >>$GITHUB_STEP_SUMMARY
+    echo "\`\`\`" >>$GITHUB_STEP_SUMMARY
+    echo "</details>" >>$GITHUB_STEP_SUMMARY
+    exit 1
+fi
+
+# -----------------
+# CREATING TEMPLATE
+# -----------------
+
+set -e # Enable exiting on error
+
+if (($template == 1)); then
+    echo "::group::Updating Makefile"
+
+    sed -i "s/^VERSION:=.*\$/VERSION:=${postfix}/" Makefile
+
+    cat Makefile
+
+    echo "::endgroup::"
+
+    echo "::group::Creating ${name} template"
+
+    make template $make_args
+
+    echo "::endgroup::"
+
+    # --------------
+    # UNZIP TEMPLATE
+    # --------------
+
+    echo "::group::Unzipping template"
+
+    unzip -o $name -d template # Unzip the template
+
+    echo "::endgroup::"
+
+fi
+
+# ---------------------------
+# ADDING VERSION, LICENSE
+# AND README TO THE TEMPLATE
+# FOLDER
+# ---------------------------
+if [[ "$INPUT_COPY_README_AND_LICENSE_TO_INCLUDE" == "true" ]]; then
+    if [[ "$INPUT_LIB_FOLDER_NAME" != "" ]]; then
+        echo "::group::Adding version, license and readme to the template folder"
+
+        echo $version >template/include/$INPUT_LIB_FOLDER_NAME/VERSION
+        find . -maxdepth 0 -type f -iname "LICENSE*" -exec cp -n {} template/include/$INPUT_LIB_FOLDER_NAME/ \;
+        find . -maxdepth 0 -type f -iname "README*" -exec cp -n {} template/include/$INPUT_LIB_FOLDER_NAME/ \;
+        echo "::endgroup::"
+    else
+        echo "::group::Adding version, license and readme to the template folder"
+        echo "Error: You must provide a library folder name if copy_readme_and_license_to_include is true" >&2
+        echo "::endgroup::"
+        # exit with an error code of 2, representing the error code for missing library folder name
+        # Redundant, but just in case
+        exit 102502 # This is the string "pros-build" turned into int values, added together, and then multiplied by 10 plus the error code at the end (error code 3). This is to hopefully avoid conflicts with other error codes.
+    fi
+fi
+# -----------
+# JOB SUMMARY
+# -----------
+norm_output=$(cat "$STD_EDITED_OUTPUT")
+echo "# ✅ Build Completed" >>$GITHUB_STEP_SUMMARY
+echo "Build completed in $build_time seconds" >>$GITHUB_STEP_SUMMARY
+echo "Total Build Script Runtime: $(($SECONDS - $script_start_time)) seconds" >>$GITHUB_STEP_SUMMARY
+echo "## 📝 Library Name: ${library_name} @ ${version}" >>$GITHUB_STEP_SUMMARY
+echo "### 🔐 SHA: ${sha}" >>$GITHUB_STEP_SUMMARY
+if (($template == 1)); then
+    echo "### 📁 Artifact Name: ${name}" >>$GITHUB_STEP_SUMMARY
+fi
+echo "***" >>$GITHUB_STEP_SUMMARY
+echo "#### 📄 Output from Make" >>$GITHUB_STEP_SUMMARY
+echo "<details><summary>Click to expand</summary>" >>$GITHUB_STEP_SUMMARY
+echo "" >>$GITHUB_STEP_SUMMARY
+echo "" >>$GITHUB_STEP_SUMMARY
+echo "\`\`\`" >>$GITHUB_STEP_SUMMARY
+echo "$norm_output" >>$GITHUB_STEP_SUMMARY
+echo "\`\`\`" >>$GITHUB_STEP_SUMMARY
+echo "</details>" >>$GITHUB_STEP_SUMMARY
```

Comment:
you can have multiline strings
```bash
echo "***
#### 📄 Output from Make
<details><summary>Click to expand</summary>
...
" >>$GITHUB_STEP_SUMMARY

Reply:
how tf did you make a multiline comment (on github)?

Reply:
do I need backslashes after each line? Last time I treid it didn't want to work.

Reply:
`````md
you can have multiline strings
```bash
echo "***
#### 📄 Output from Make
<details><summary>Click to expand</summary>
...
" >>$GITHUB_STEP_SUMMARY
`````

Reply:
no I meant your selection from line 215 to 223

Reply:
oh, i dragged

Reply:
that never works for me :(